app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -904,6 +918,115 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return super.onOptionsItemSelected(item);
   }
 
+  /** Dialog to take user confirmation before deleting all notes */
+  private void showClearAllNotesDialog() {
+
+    AlertDialog.Builder builder;
+    if (sharedPreferenceUtil != null && sharedPreferenceUtil.nightMode()) { // Night Mode support
+      builder = new AlertDialog.Builder(this, R.style.AppTheme_Dialog_Night);
+    } else {
+      builder = new AlertDialog.Builder(this);
+    }
+
+    builder.setMessage(this.getString(R.string.delete_notes_confirmation_msg))
+        .setNeutralButton(this.getString(R.string.dismiss), new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) {
+            // Do nothing
+          }
+        })
+        .setPositiveButton(this.getString(R.string.delete), new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) {
+            clearAllNotes();
+          }
+        })
+        .show();
+  }
+
+  /** Method to delete all user notes */
+  private void clearAllNotes() {
+
+    boolean result = true; // Result of all delete() calls is &&-ed to this variable
+
+    if(AddNoteDialog.isExternalStorageWritable()) {
+      if(ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+        Log.d("MainActivity", "WRITE_EXTERNAL_STORAGE permission not granted");
+        Toast.makeText(this, this.getString(R.string.ext_storage_permission_not_granted), Toast.LENGTH_LONG);
+        return;
+      }
+
+      File notesDirectory = new File(Environment.getExternalStorageDirectory() + "/Kiwix/Notes/");
+      File[] filesInNotesDirectory = notesDirectory.listFiles();
```

Comment:
Kotlin has a very nice file.deleteRecursively(), how about we flag this method with a //TODO and revisit it at some point down the line

Reply:
[Flagged](https://github.com/kiwix/kiwix-android/pull/1198/commits/873eb2bce86c76b11962367701cb66bdf66b9f5e#diff-8f999695c436a978371f898c45907cdaR955)